src/qibo/models/error_mitigation.py

```diff
@@ -330,6 +330,7 @@ def _curve_fit(
     if backend.name == "pytorch":
         # pytorch has some problems with the `scipy.optim.curve_fit` function
         # thus we use a `torch.optim` optimizer
+        params = params.requires_grad_(True)
```

Comment:
Should this be forced here without user input?

Reply:
It can be removed but this is a function to train the parameters, this way the training will work even if the user forgets to activate the gradients. I think it is just a way to help the user.

Reply:
Let's hear @BrunoLiegiBastonLiegi

Reply:
This is not a function the user is supposed to face in principle, I mean it doesn't harm to enforce it and shouldn't be expensive, but I wonder where the error was coming from. I guess inside the mitigation functions, and thus I would say it's better to correct there.